tests/Parser/ParserTest.php

```diff
@@ -0,0 +1,125 @@
+<?php
+
+declare(strict_types=1);
+namespace DevCommunityDE\CodeFormatter\Test;
+
+use DevCommunityDE\CodeFormatter\Parser\ElemNode;
+use DevCommunityDE\CodeFormatter\Parser\Node;
+use DevCommunityDE\CodeFormatter\Parser\Parser;
+use DevCommunityDE\CodeFormatter\Parser\TextNode;
+use PHPUnit\Framework\TestCase;
+
+final class ParserTest extends TestCase
+{
+    private function parseFileToArray(string $input)
+    {
+        $parser = new Parser();
+        return iterator_to_array($parser->parseFile($input));
+    }
+
+    private function parseTextToArray(string $input)
+    {
+        $parser = new Parser();
+        return iterator_to_array($parser->parseText($input));
+    }
+
+    public function testParsingAFileBehavesTheSameAsParsingAText()
+    {
+        $input = __DIR__ . '/input-0.txt';
+        $fromText = $this->parseTextToArray(file_get_contents($input));
+        $fromFile = $this->parseFileToArray($input);
+        $this->assertEquals($fromText, $fromFile);
+    }
+
+    public function testPlainBbcodeConsumesOtherBbcodes()
+    {
+        $nodes = $this->parseTextToArray('before plain bbcode[plain][code]test[/code][/plain]');
+        $this->assertCount(2, $nodes);
+        $this->assertInstanceof(TextNode::class, $nodes[0]);
+        $this->assertEquals(Node::KIND_TEXT, $nodes[0]->getKind());
+        $this->assertEquals('before plain bbcode', $nodes[0]->getBody());
+        $this->assertInstanceof(ElemNode::class, $nodes[1]);
+        $this->assertNotTrue($nodes[1]->isCode());
+        $this->assertEquals('[code]test[/code]', $nodes[1]->getBody());
+    }
+
+    public function testBracketsInsideTextAreHandledCorrectly()
+    {
+        $nodes = $this->parseTextToArray('hello[[[[[');
+        $this->assertCount(2, $nodes);
+        $this->assertInstanceof(TextNode::class, $nodes[0]);
+        $this->assertEquals('hello', $nodes[0]->getBody());
+        $this->assertInstanceof(TextNode::class, $nodes[1]);
+        $this->assertEquals('[[[[[', $nodes[1]->getBody());
```

Comment:
I wonder why you chose to interpret this as two different text nodes rather than just a single one.

Reply:
The current algorithm uses `[` to split text-nodes, because each `[` could start a new bbcode. I'm not satisfied with this either, that's why the pr is still a draft :)

Reply:
Fixed in https://github.com/dev-community-de/code-formatter/pull/36/commits/0f57d541b345c4b6c82e676eb185019c2e36cf40